language/tools/move-cli/src/sandbox/cli.rs

```diff
@@ -0,0 +1,284 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    base, sandbox, sandbox::utils::Mode, shared, Move, NativeFunctionRecord, DEFAULT_SOURCE_DIR,
+};
+use anyhow::Result;
+use move_core_types::{
+    errmap::ErrorMapping, language_storage::TypeTag, parser,
+    transaction_argument::TransactionArgument,
+};
+use std::{fs, path::Path};
+use structopt::StructOpt;
+
+use super::utils::on_disk_state_view::OnDiskStateView;
+
+#[derive(StructOpt)]
+pub enum SandboxCommand {
+    /// Compile the specified modules and publish the resulting bytecodes in global storage
+    #[structopt(name = "publish")]
+    Publish {
+        /// The source files containing modules to publish
+        #[structopt(
+            name = "PATH_TO_SOURCE_FILE",
+            default_value = DEFAULT_SOURCE_DIR,
+        )]
+        source_files: Vec<String>,
+        /// If set, fail during compilation when attempting to publish a module that already
+        /// exists in global storage
+        #[structopt(long = "no-republish")]
+        no_republish: bool,
+        /// By default, code that might cause breaking changes for bytecode
+        /// linking or data layout compatibility checks will not be published.
+        /// Set this flag to ignore breaking changes checks and publish anyway
+        #[structopt(long = "ignore-breaking-changes")]
+        ignore_breaking_changes: bool,
+        /// fix publishing order
+        #[structopt(short = "m", long = "override-ordering")]
```

Comment:
Nit-pick: personally I prefer not to have short option names if there's no clear single letter that makes sense for the option. In this case, `m` doesn't appear in the name `override-ordering`, so I wouldn't make it available as a short name.